redisinsight/api/src/modules/tag/dto/update-tag.dto.ts

```diff
@@ -0,0 +1,24 @@
+import { ApiPropertyOptional } from '@nestjs/swagger';
+import { IsOptional, IsString, Length, Matches } from 'class-validator';
+
+export class UpdateTagDto {
+  @ApiPropertyOptional({
+    description: 'Key of the tag.',
+    type: String,
+  })
+  @IsOptional()
+  @IsString()
+  @Length(1, 64)
+  @Matches(/^[a-zA-Z0-9\-_\.@:+ ]+$/)
+  key?: string;
+
+  @ApiPropertyOptional({
+    description: 'Value of the tag.',
+    type: String,
+  })
+  @IsOptional()
+  @IsString()
+  @Length(1, 128)
+  @Matches(/^[a-zA-Z0-9\-_\.@:+ ]+$/)
+  value?: string;
+}
```

Comment:
There is no need to define all properties here. The main difference is that fields became optional. Let's use `PartialType` here. We can combine it with `OmitType`, `PickType` in the future if DTOs should be slightly different